### PR TITLE
Upgrade 1.3 to 1.4

### PR DESCRIPTION
**Pull request by fiammybe**
Upgrade a 1.3.11 to a 1.4.0 system

### CODE REVIEW

File: htdocs/modules/system/admin/banners/banners.php [hunk 1]
```diff
@@ -1,393 +1,393 @@
...
+		. _AM_CLINAMET . "<input type='text' name='name' value='$name' size='30' maxlength='60' /><br />"
+		. _AM_CONTNAMET . "<input type='text' name='contact' value='$contact' size='30' maxlength='60' /><br />"
+		. _AM_CONTMAILT . "<input type='text' name='email' size='30' maxlength='60' value='$email' /><br />"
+		. _AM_CLILOGINT . "<input type='text' name='login' size='12' maxlength='10' value='$login' /><br />"
+		. _AM_CLIPASST . "<input type='text' name='passwd' size='12' maxlength='10' value='$passwd' /><br />"
+		. _AM_EXTINFO . "<br /><textarea name='extrainfo' cols='60' rows='10' />$extrainfo</textarea><br />"
+		. "<input type='hidden' name='cid' value='$cid' />"
+		. "<input type='hidden' name='op' value='BannerClientChange' />"
+		. icms::$security->getTokenHTML()
+		. "<input type='hidden' name='fct' value='banners' /><input type='submit' value='" . _AM_CHGCLI . "' />";
```
[the remark this answers is not in the document]
Comment (fiammybe) [reply htdocs/modules/system/admin/banners/banners.php]: file will be removed

File: htdocs/modules/system/admin/banners/banners.php [hunk 2]
```diff
@@ -1,393 +1,393 @@
...
+		$extrainfo = icms_core_DataFilter::htmlSpecialChars($extrainfo);
+		echo "<table width='100%' border='0' cellspacing='1' class='outer'><tr><td class=\"odd\">";
+		echo "<h4>" . _AM_EDITADVCLI . "</h4><form action='admin.php' method='post'>"
+		. _AM_CLINAMET . "<input type='text' name='name' value='$name' size='30' maxlength='60' /><br />"
+		. _AM_CONTNAMET . "<input type='text' name='contact' value='$contact' size='30' maxlength='60' /><br />"
+		. _AM_CONTMAILT . "<input type='text' name='email' size='30' maxlength='60' value='$email' /><br />"
+		. _AM_CLILOGINT . "<input type='text' name='login' size='12' maxlength='10' value='$login' /><br />"
+		. _AM_CLIPASST . "<input type='text' name='passwd' size='12' maxlength='10' value='$passwd' /><br />"
+		. _AM_EXTINFO . "<br /><textarea name='extrainfo' cols='60' rows='10' />$extrainfo</textarea><br />"
+		. "<input type='hidden' name='cid' value='$cid' />"
```
[the remark this answers is not in the document]
Comment (fiammybe) [reply htdocs/modules/system/admin/banners/banners.php]: file will be removed

File: htdocs/modules/system/admin/banners/banners.php [hunk 3]
```diff
@@ -1,393 +1,393 @@
...
+		$passwd = icms_core_DataFilter::htmlSpecialChars($passwd);
+		$extrainfo = icms_core_DataFilter::htmlSpecialChars($extrainfo);
+		echo "<table width='100%' border='0' cellspacing='1' class='outer'><tr><td class=\"odd\">";
+		echo "<h4>" . _AM_EDITADVCLI . "</h4><form action='admin.php' method='post'>"
+		. _AM_CLINAMET . "<input type='text' name='name' value='$name' size='30' maxlength='60' /><br />"
+		. _AM_CONTNAMET . "<input type='text' name='contact' value='$contact' size='30' maxlength='60' /><br />"
+		. _AM_CONTMAILT . "<input type='text' name='email' size='30' maxlength='60' value='$email' /><br />"
+		. _AM_CLILOGINT . "<input type='text' name='login' size='12' maxlength='10' value='$login' /><br />"
+		. _AM_CLIPASST . "<input type='text' name='passwd' size='12' maxlength='10' value='$passwd' /><br />"
+		. _AM_EXTINFO . "<br /><textarea name='extrainfo' cols='60' rows='10' />$extrainfo</textarea><br />"
```
[the remark this answers is not in the document]
Comment (fiammybe) [reply htdocs/modules/system/admin/banners/banners.php]: file will be removed

File: htdocs/modules/system/admin/banners/banners.php [hunk 4]
```diff
@@ -1,393 +1,393 @@
...
+		$login = icms_core_DataFilter::htmlSpecialChars($login);
+		$passwd = icms_core_DataFilter::htmlSpecialChars($passwd);
+		$extrainfo = icms_core_DataFilter::htmlSpecialChars($extrainfo);
+		echo "<table width='100%' border='0' cellspacing='1' class='outer'><tr><td class=\"odd\">";
+		echo "<h4>" . _AM_EDITADVCLI . "</h4><form action='admin.php' method='post'>"
+		. _AM_CLINAMET . "<input type='text' name='name' value='$name' size='30' maxlength='60' /><br />"
+		. _AM_CONTNAMET . "<input type='text' name='contact' value='$contact' size='30' maxlength='60' /><br />"
+		. _AM_CONTMAILT . "<input type='text' name='email' size='30' maxlength='60' value='$email' /><br />"
+		. _AM_CLILOGINT . "<input type='text' name='login' size='12' maxlength='10' value='$login' /><br />"
+		. _AM_CLIPASST . "<input type='text' name='passwd' size='12' maxlength='10' value='$passwd' /><br />"
```
[the remark this answers is not in the document]
Comment (fiammybe) [reply htdocs/modules/system/admin/banners/banners.php]: file will be removed

File: htdocs/modules/system/admin/banners/banners.php [hunk 5]
```diff
@@ -1,393 +1,393 @@
...
+			$result = icms::$xoopsDB->query("SELECT cid, name FROM " . icms::$xoopsDB->prefix("bannerclient"));
+			while (list($cid, $name) = icms::$xoopsDB->fetchRow($result)) {
+				$name = icms_core_DataFilter::htmlSpecialChars($name);
+				echo "<option value='$cid'>$name</option>";
+			}
+			echo "</select><br />"
+			. _AM_IMPPURCHT . "<input type='text' name='imptotal' size='12' maxlength='11' /> " . icms_conv_nr2local(0) . " = " . _AM_UNLIMIT . "<br />"
+			. _AM_IMGURLT . "<input type='text' name='imageurl' size='50' maxlength='255' /><br />"
+			. _AM_CLICKURLT . "<input type='text' name='clickurl' size='50' maxlength='255' /><br />"
+			. _AM_USEHTML . " <input type='checkbox' name='htmlbanner' value='1' /><br />"
```
[the remark this answers is not in the document]
Comment (fiammybe) [reply htdocs/modules/system/admin/banners/banners.php]: file will be removed

File: htdocs/modules/system/admin/banners/banners.php [hunk 6]
```diff
@@ -1,393 +1,393 @@
...
+			. _AM_CLINAMET . "<select name='cid'>";
+			$result = icms::$xoopsDB->query("SELECT cid, name FROM " . icms::$xoopsDB->prefix("bannerclient"));
+			while (list($cid, $name) = icms::$xoopsDB->fetchRow($result)) {
+				$name = icms_core_DataFilter::htmlSpecialChars($name);
+				echo "<option value='$cid'>$name</option>";
+			}
+			echo "</select><br />"
+			. _AM_IMPPURCHT . "<input type='text' name='imptotal' size='12' maxlength='11' /> " . icms_conv_nr2local(0) . " = " . _AM_UNLIMIT . "<br />"
+			. _AM_IMGURLT . "<input type='text' name='imageurl' size='50' maxlength='255' /><br />"
+			. _AM_CLICKURLT . "<input type='text' name='clickurl' size='50' maxlength='255' /><br />"
```
[the remark this answers is not in the document]
Comment (fiammybe) [reply htdocs/modules/system/admin/banners/banners.php]: file will be removed

File: htdocs/modules/system/admin/banners/banners.php [hunk 7]
```diff
@@ -1,393 +1,393 @@
...
+			. "</h4><form action='admin.php' method='post'>"
+			. _AM_CLINAMET . "<select name='cid'>";
+			$result = icms::$xoopsDB->query("SELECT cid, name FROM " . icms::$xoopsDB->prefix("bannerclient"));
+			while (list($cid, $name) = icms::$xoopsDB->fetchRow($result)) {
+				$name = icms_core_DataFilter::htmlSpecialChars($name);
+				echo "<option value='$cid'>$name</option>";
+			}
+			echo "</select><br />"
+			. _AM_IMPPURCHT . "<input type='text' name='imptotal' size='12' maxlength='11' /> " . icms_conv_nr2local(0) . " = " . _AM_UNLIMIT . "<br />"
+			. _AM_IMGURLT . "<input type='text' name='imageurl' size='50' maxlength='255' /><br />"
```
[the remark this answers is not in the document]
Comment (fiammybe) [reply htdocs/modules/system/admin/banners/banners.php]: file will be removed

File: htdocs/modules/system/admin/banners/banners.php [hunk 8]
```diff
@@ -1,393 +1,393 @@
...
+			echo "<h4>" . _AM_ADDNWBNR
+			. "</h4><form action='admin.php' method='post'>"
+			. _AM_CLINAMET . "<select name='cid'>";
+			$result = icms::$xoopsDB->query("SELECT cid, name FROM " . icms::$xoopsDB->prefix("bannerclient"));
+			while (list($cid, $name) = icms::$xoopsDB->fetchRow($result)) {
+				$name = icms_core_DataFilter::htmlSpecialChars($name);
+				echo "<option value='$cid'>$name</option>";
+			}
+			echo "</select><br />"
+			. _AM_IMPPURCHT . "<input type='text' name='imptotal' size='12' maxlength='11' /> " . icms_conv_nr2local(0) . " = " . _AM_UNLIMIT . "<br />"
```
[the remark this answers is not in the document]
Comment (fiammybe) [reply htdocs/modules/system/admin/banners/banners.php]: file will be removed

File: htdocs/modules/system/admin/banners/banners.php [hunk 9]
```diff
@@ -1,393 +1,393 @@
...
+		echo "</td></tr></table>";
+		echo "<br />";
+		// Add Banner
+		$result = icms::$xoopsDB->query("SELECT COUNT(*) FROM " . icms::$xoopsDB->prefix("bannerclient"));
+		list($numrows) = icms::$xoopsDB->fetchRow($result);
+		if ($numrows > 0) {
+			echo"<table width='100%' border='0' cellspacing='1' class='outer'><tr><td class=\"odd\">";
+			echo "<h4>" . _AM_ADDNWBNR
+			. "</h4><form action='admin.php' method='post'>"
+			. _AM_CLINAMET . "<select name='cid'>";
```
[the remark this answers is not in the document]
Comment (fiammybe) [reply htdocs/modules/system/admin/banners/banners.php]: file will be removed

File: htdocs/install/makedata.php [hunk 10]
```diff
@@ -85,11 +85,6 @@ function make_data(&$dbm, &$cm, $adminname, $adminlogin_name, $adminpass, $admin
 		. ", (0,".$gruops['XOOPS_GROUP_ADMIN'].",1,1,'content_admin')"
 		. ", (0,".$gruops['XOOPS_GROUP_ADMIN'].",1,1,'use_wysiwygeditor')");
 
-	// data for table 'banner'
-	$dbm->insert("banner", " (bid, cid, imptotal, impmade, clicks, imageurl, clickurl, date, htmlcode) VALUES (1, 1, 0, 1, 0, '".XOOPS_URL."/images/banners/impresscms_banner.gif', '"._INSTALL_LOCAL_SITE."', 1008813250, '')"
```
Comment (MekDrop) [inline htdocs/install/makedata.php]: I think you should remove related files to. Doesn't make sense to have banner folder that is not used anymore.

File: htdocs/modules/system/include/update-13.php [hunk 11]
```diff
@@ -0,0 +1,392 @@
...
+
+	/* Change formtype to textsarea for Protector values */
+	$sql = "UPDATE `" . $table->name() . "` SET conf_formtype = 'textsarea' WHERE conf_formtype = 'textarea' AND conf_title LIKE '_MI_PROTECTOR%';";
+	$icmsDatabaseUpdater->runQuery($sql, sprintf(_DATABASEUPDATER_MSG_QUERY_SUCCESSFUL, $sql), sprintf(_DATABASEUPDATER_MSG_QUERY_FAILED, $sql));
+
+	/* Change formtype to textbox for Protector values */
+	$sql = "UPDATE `" . $table->name() . "` SET conf_formtype = 'textbox' WHERE conf_formtype = 'text' AND conf_title LIKE '_MI_PROTECTOR%';";
+	$icmsDatabaseUpdater->runQuery($sql, sprintf(_DATABASEUPDATER_MSG_QUERY_SUCCESSFUL, $sql), sprintf(_DATABASEUPDATER_MSG_QUERY_FAILED, $sql));
+
+
```
Comment (MekDrop) [inline htdocs/modules/system/include/update-13.php]: why here too many empty lines?

Comment (fiammybe) [reply htdocs/modules/system/include/update-13.php]: Removed the empty linkes, you're right that there were way too many :-)

File: htdocs/install/makedata.php (hunk 10 above)
Comment (fiammybe) [reply htdocs/install/makedata.php]: Normally the content of the banner folder, as well as the folder, is removed as well (update.php line 117 - 121).

**Pull request merged**